metals/src/main/scala/scala/meta/internal/metals/callHierarchy/CallHierarchyItemBuilder.scala

```diff
@@ -0,0 +1,161 @@
+package scala.meta.internal.metals.callHierarchy
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.BuildTargets
+import scala.meta.internal.metals.Compilers
+import scala.meta.internal.metals.HoverExtParams
+import scala.meta.internal.metals.Icons
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+
+import org.eclipse.{lsp4j => l}
+
+private[callHierarchy] final class CallHierarchyItemBuilder(
+    workspace: AbsolutePath,
+    icons: Icons,
+    compilers: () => Compilers,
+    buildTargets: BuildTargets,
+) {
+  private def buildItem(
+      name: String,
+      kind: l.SymbolKind,
+      uri: String,
+      range: l.Range,
+      selectionRange: l.Range,
+      detail: String,
+      data: CallHierarchyItemInfo,
+  ): l.CallHierarchyItem = {
+    val item = new l.CallHierarchyItem(name, kind, uri, range, selectionRange)
+    item.setDetail(detail)
+    item.setData(data)
+    item
+  }
+
+  // When we search the call hierarchy of a class, we also need to search the symbol of the primary constructor
+  private def getContructorRelatedSymbol(
+      info: s.SymbolInformation
+  ): Array[String] =
+    info.signature match {
+      case s.ClassSignature(_, _, _, declarations) =>
+        val constructor = declarations
+          .flatMap(
+            _.symlinks.find(_.endsWith("`<init>`()."))
+          )
+        // for case class apply is like a constructor
+        val applySymbol = Option.when(info.isCase)(
+          info.symbol.replace("#", ".apply().")
+        )
+        Set(Some(info.symbol), constructor, applySymbol).flatten.toArray
+      case _ =>
+        Array(info.symbol)
+    }
+
+  /**
+   * Get the display name from the symbol information.
+   * For constructor, strip the symbol to the class name,
+   * otherwise return symbols's display name.
+   */
+  private def getName(info: s.SymbolInformation): String =
+    if (info.isConstructor && info.displayName == "<init>")
+      info.symbol.slice(
+        info.symbol.lastIndexOf("/") + 1,
+        info.symbol.lastIndexOf("#"),
+      )
+    else info.displayName
+
+  // Regular expression to capture the group inside hovers that indicate the symbol signature.
+  private val symbolSignatureExtractor =
+    """Symbol signature\*\*:\n```scala\n(.*)\n```""".r
```

Comment:
We could potentially have an additional method or just boolean option for presentation compiler to just get the signature, which would not require matching here.

Reply:
Yeah I will try, this was not my first choice because the method that handle that is almost 100 lines of codes and a little bit hard to understand, but I will dive on it

Reply:
If the problem is the additional docs + expression type then it should be simple to disable with a flag